### PR TITLE
Restructure motivation/design and add informal summary

### PR DESCRIPTION
**Pull request by sipa**
This aims to address #138.

### CODE REVIEW

**Comment by ajtowns** [comment]
ACK based on quick skim

**Comment by AdamISZ** [comment]
LGTM. Liked the `Q` part, that's pretty much exactly what I had in mind.

**Comment by sipa** [comment]
@AdamISZ Done.

Also added a short rationale on security assumptions with a link to @apoelstra's proof.

File: bip-taproot.mediawiki [hunk 1]
```diff
@@ -41,6 +41,7 @@ As a result we choose this combination of technologies:
...
 
+Informally, the resulting design is as follows. A new witness version is added (1), whose programs consist of 32-byte encodings of points ''Q''. ''Q'' is computed as ''P + hash(P||m)G'' for a public key ''P'', and the root ''m'' of a Merkle tree whose leaves consist of a version number and a script. These outputs can be spent directly by providing a signature for ''Q'', or indirectly by revealing ''P'', the script and leaf version, inputs that satisfy the script, and a Merkle branch that proves ''Q'' committed to that leaf. All hashes in this construction (the hash for computing ''Q'' from ''P'', the hashes inside the Merkle tree's inner nodes, and the signature hashes used) are tagged to guarantee domain separation.
```
Comment (ajtowns) [inline bip-taproot.mediawiki]: "and a Merkle **path**" rather than branch?

**Comment by ajtowns** [comment]
LGTM

File: bip-taproot.mediawiki (hunk 1 above)
Comment (sipa) [reply bip-taproot.mediawiki]: Fixed.

File: bip-taproot.mediawiki [hunk 2]
```diff
@@ -45,7 +41,7 @@ As a result we choose this combination of technologies:
...
 
-Not included in this proposal are additional features like new sighash modes or opcodes that can be included with no loss in effectiveness as a future extension. Also not included is cross-input aggregation, as it [https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-March/015838.html interacts] in complex ways with upgrade mechanisms and solutions to that are still [https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-October/016461.html in flux].
+Informally, the resulting design is as follows. A new witness version is added (1), whose programs consist of 32-byte encodings of points ''Q''. ''Q'' is computed as ''P + hash(P||m)G'' for a public key ''P'', and the root ''m'' of a Merkle tree whose leaves consist of a version number and a script. These outputs can be spent directly by providing a signature for ''Q'', or indirectly by revealing ''P'', the script and leaf version, inputs that satisfy the script, and a Merkle path that proves ''Q'' committed to that leaf. All hashes in this construction (the hash for computing ''Q'' from ''P'', the hashes inside the Merkle tree's inner nodes, and the signature hashes used) are tagged to guarantee domain separation.
```
Comment (instagibbs) [inline bip-taproot.mediawiki]: s/as follows./as follows:/

Comment (jonatack) [inline bip-taproot.mediawiki]: *A new witness version is added (1),*... apologies if I am misunderstanding; does the `(1)` refer to the size in bytes? If yes, "(size in bytes)" is defined much further down (and appears to be missing an "s", e.g. s/(size in byte)/(size in bytes)/ unless it is a convention I am unfamiliar with).

Comment (sipa) [reply bip-taproot.mediawiki]: No, the new version number is 1.

Comment (sipa) [reply bip-taproot.mediawiki]: Done.

Comment (sipa) [reply bip-taproot.mediawiki]: Fixed, changed it to "(version 1)".

**Pull request merged**